source/openqasm/openqasm3/printer.py

```diff
@@ -0,0 +1,700 @@
+import contextlib
+import dataclasses
+import io
+
+from typing import Sequence, Optional
+
+from . import ast, properties
+from .visitor import QASMVisitor
+
+__all__ = ("dump", "dumps")
+
+
+def dump(node: ast.QASMNode, file: io.TextIOBase, **kwargs):
+    _Printer(file, **kwargs).visit(node)
+
+
+def dumps(node: ast.QASMNode, **kwargs):
+    out = io.StringIO()
+    _Printer(out, **kwargs).visit(node)
```

Comment:
Optional suggestion:
```suggestion
    dump(node, out, **kwargs)
```
This will allow the tests to cover `dump()`. Also, if we ever need to change the interface of `_Printer`, we don't have to change it multiple places.

Reply:
Done in 09bbb86.